pkg/auth/context/context.go

```diff
@@ -0,0 +1,21 @@
+package auth
+
+import "context"
+
+// SAAuthenticatedContextKey is the context key for the SAAuthenticated flag.
+var SAAuthenticatedContextKey = struct{}{}
+
+// IsSAAuthenticated returns true if the SAAuthenticated flag is set in the context.
+func IsSAAuthenticated(ctx context.Context) bool {
+	authed, ok := ctx.Value(SAAuthenticatedContextKey).(bool)
+	if !ok {
+		return false
+	}
+
+	return authed
+}
+
+// SetSAAuthenticated sets the SAAuthenticated flag in the context.
+func SetSAAuthenticated(ctx context.Context) context.Context {
+	return context.WithValue(ctx, SAAuthenticatedContextKey, true)
+}
```

Comment:
Not sure if needed but since this is a `SetXXX` we could add a parameter to the signature:

```go
func SetSAAuthenticated(ctx context.Context, authenticated bool) context.Context {
	return context.WithValue(ctx, SAAuthenticatedContextKey, authenticated)
}
```

Reply:
In our case, we're never setting it to anything but true, so we feel like the parameter is unnecessary.